src/com/google/enterprise/adaptor/DocIdSender.java

```diff
@@ -337,16 +411,28 @@ public GroupPrincipal pushGroupDefinitions(
       double mean = ((double) numMembers) / numGroups;
       log.finer("mean size of groups: " + mean);
     }
+
+    // Remember the latest target group source fed. If we just did a
+    // full group feed, delete the previous target group source entries.
+    String oldGroupSource = groupSources.put(groupSource, feedSourceName);
+    if (feedType == FULL && (oldGroupSource != null)) {
+      // Delete any entries from the older group source by pushing an
+      // empty full feed.
+      log.log(Level.FINE, "Deleting entries from previous group source {0}",
+          oldGroupSource);
+      List<Map.Entry<GroupPrincipal, T>> emptyList = Collections.emptyList();
+      pushSizedBatchOfGroups(emptyList, caseSensitive, FULL, oldGroupSource,
```

Comment:
With current imple if client does:
1. incremental feeds to "foo"
2. full push to "foo" (which is full push to "foo-FULL1")
then "foo" gets deleted.

That seems like what Igor was suggesting. I don't follow the comment then that incremental should use the name of last FULL feed pushed.  Does that mean, that if client (3) pushes more incremental to "foo" those foo will go to "foo-FULL1"?